lib/rspec/mocks/any_instance/stub_chain.rb

```diff
@@ -30,7 +30,7 @@ def invocation_order
             :with => [nil],
             :and_return => [:with, nil],
             :and_raise => [:with, nil],
-            :and_yield => [:with, nil],
+            :and_yield => [:with, :and_yield, nil],
```

Comment:
Not quite, as `and_return`,  `and_raise`, and `and_call_original` or `and_wrap_original` should all be mutually exclusive I think.